### ISSUE TITLE
Tracking Progress with the progress event

### ISSUE THREAD

**Issue by DrYSG** [archiverjs/node-archiver#234]
Based on issues listed here, and testing. I see that after finalize() has been called, archive.on("entry") will get an event every time a file is processed (whether it is from .file() .glob() or .directory() [this should be called out int he documentation]

But if I want a progress returned to the user, I have to know how many files (entries) have been found when the .finalize() call is made. Is there a way to do this? I don't see this info in the 'entry' event. But I might be missing something.

**Comment by soyuka** [comment]
Extract from one of my code (archive is an `archiver('zip')`):

```
          let t = archive._entries.reduce((res, a) => {
            res += a.stats.size
            return res
          }, 0)

          let percent = 100 - ((totalSize - t) / totalSize) * 100
          let s = archive.pointer()
          channel.send('data', {type: 'archive:progress', data: {path: file, size: s, progress: percent, total: totalSize, prettySize: prettyBytes(s)}})
```

Hope this helps.

**Comment by DrYSG** [comment]
@soyuka Not quite clear where you are getting totalSize from. Some of my folders are filled by a prior step in express, via  a childProcesss that tiles a map, So I don't know a priory the fileCount or fileSizes of the folder. When I looked at archive._entries after the finalize, the length was 0. I think I can get by with a file Count proress percent, and don't have to go with the archive.pointer() byte count. Can you give me a tip here?

**Comment by soyuka** [comment]
Sorry, edited just a few minutes ago. 

I indeed am computing the directory size first. You can achieve that by recursively `stat` the directory. Archiver doesn't know the full size until the end because it's adding files while reading. Though with `_entries` you can find the current stored size.
`pointer()` represents the `archive` size in bytes.

If we were able to know the compression rate before starting, we could reverse the process and compute a progress without the need of the total size. I don't think that's possible because it may vary between files.

**Comment by DrYSG** [comment]
given that it is a hierarchical folder, do you have any suggestion better than using a tree walker package such as this: https://www.npmjs.com/package/get-folder-size

**Comment by soyuka** [comment]
See #253 
Closing in favor of #245
